src/shared/fork-journal.c

```diff
@@ -0,0 +1,203 @@
+                "journalctl",
+                "--follow",
+                "--no-pager",
+                "--lines=1",
+                scope == RUNTIME_SCOPE_SYSTEM ? "--unit" : "--user-unit",
+                unit,
+                NULL,
+        };
+
+        if (DEBUG_LOGGING) {
+                _cleanup_free_ char *l = quote_command_line((char**) argv, SHELL_ESCAPE_EMPTY);
+                log_debug("Invoking '%s' as child.", strnull(l));
+        }
+
+        BLOCK_SIGNALS(SIGCHLD);
+
+        _cleanup_(pidref_done_sigkill_wait) PidRef child = PIDREF_NULL;
+        r = pidref_safe_fork_full(
+                        "(journalctl)",
+                        (const int[3]) { -EBADF, STDOUT_FILENO, STDERR_FILENO },
+                        /* except_fds= */ NULL,
+                        /* n_except_fds= */ 0,
+                        FORK_RESET_SIGNALS|FORK_DEATHSIG_SIGTERM|FORK_CLOSE_ALL_FDS|FORK_RLIMIT_NOFILE_SAFE|FORK_REARRANGE_STDIO,
+                        &child);
+        if (r < 0)
+                return r;
+        if (r == 0) {
+                /* In the child: */
+
+                if (setenv("NOTIFY_SOCKET", addr_string, /* overwrite= */ true) < 0) {
+                        log_debug_errno(errno, "Failed to set $NOTIFY_SOCKET: %m");
+                        _exit(EXIT_MEMORY);
+                }
+
+                r = invoke_callout_binary(argv[0], (char**) argv);
+                log_debug_errno(r, "Failed to invoke journalctl: %m");
+                _exit(EXIT_EXEC);
+        }
+
+        _cleanup_(sd_event_unrefp) sd_event *event = NULL;
+        r = sd_event_new(&event);
+        if (r < 0)
+                return r;
+
+        _cleanup_(sd_event_source_disable_unrefp) sd_event_source *child_event_source = NULL;
+        r = event_add_child_pidref(event, &child_event_source, &child, WEXITED, on_child_exit, &child);
+        if (r < 0)
+                return r;
+
+        _cleanup_(sd_event_source_disable_unrefp) sd_event_source *notify_event_source = NULL;
+        r = sd_event_add_io(event, &notify_event_source, notify_fd, EPOLLIN, on_child_notify, &child);
+        if (r < 0)
+                return r;
+
+        /* We want the notification message from the child before the SIGCHLD */
+        r = sd_event_source_set_priority(notify_event_source, SD_EVENT_PRIORITY_NORMAL-10);
+        if (r < 0)
+                return r;
+
+        r = sd_event_loop(event);
+        if (r < 0)
+                return r;
+        assert(r == 0);
+
+        if (pids) {
+                _cleanup_(pidref_freep) PidRef *copy = NULL;
+
+                r = pidref_dup(&child, &copy);
+                if (r < 0)
+                        return r;
+
+                r = set_ensure_consume(pids, &pidref_hash_ops_free, TAKE_PTR(copy));
+                if (r < 0)
+                        return r;
+        }
+
+        pidref_done(&child); /* Disarm auto-kill */
+
+        return 0;
+}
+
+Set *journal_terminate(Set *pids) {
```

Comment:
Please swap the space and `*`:
`Set *journal_terminate()` -> `Set* journal_terminate()`